### PR TITLE
[zh-cn] init translation of 'UI pseudo classes'

### PR DESCRIPTION
**Pull request by jasonren0403**
### Motivation

reduce LearnSidebar flaws by one :(

### CODE REVIEW

File: files/zh-cn/learn/forms/ui_pseudo-classes/index.md [hunk 1]
```diff
@@ -0,0 +1,620 @@
...
+- {{cssxref(":hover")}}：只在鼠标指针悬停在一个元素上时选择该元素。
+- {{cssxref(":focus")}}：只在元素被聚焦时选择该元素（也就是说，通过键盘上的 tab 键选中该元素）。
+- {{cssxref(":active")}}：只在元素被激活时选择该元素（也就是说，通过点击或键盘上的 <kbd>Return</kbd> / <kbd>Enter</kbd> 键选中该元素）。
+
+这些基本的伪类现在对你来说应该很熟悉了。最近，[CSS Selector Level 3](https://www.w3.org/TR/selectors-3/) 和 [CSS Basic UI Level 3](https://drafts.csswg.org/css-ui-3/#pseudo-classes) 增加了更多与 HTML 表单有关的伪类，提供了其他几个有用的定位条件以便于利用。我们将在下面的章节中更详细地讨论这些内容，但简单地说，我们要看的主要内容是：
+
+- {{cssxref(':required')}} 和 {{cssxref(':optional')}}：针对必需的或可选的表单控件。
+- {{cssxref(":valid")}} 和 {{cssxref(":invalid")}}、{{cssxref(":in-range")}} 和 {{cssxref(":out-of-range")}}：针对表单控件，根据对其设置的表单验证约束，或范围内/范围外，是有效/无效的。
+- {{cssxref(":enabled")}} 和 {{cssxref(":disabled")}}、{{cssxref(":read-only")}} 和 {{cssxref(":read-write")}}：针对启用或禁用的表单控件（例如，设置了 `disabled` HTML 属性），以及读写或只读的表单控件（例如，设置了 `readonly` HTML 属性）。
+- {{cssxref(":checked")}}、{{cssxref(":indeterminate")}} 和 {{cssxref(":default")}}：分别针对被选中的复选框和单选按钮，处于不确定的状态（既不被选中也不被选中），以及页面加载时默认选择的选项（例如，一个设置了 `checked` 属性的 [`<input type="checkbox">`](/zh-CN/docs/Web/HTML/Element/input/checkbox)，或者一个设置了 `selected` 属性[`<option>`](/zh-CN/docs/Web/Element/option) 元素）。
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/ui_pseudo-classes/index.md:43]:
```suggestion
- {{cssxref(":checked")}}、{{cssxref(":indeterminate")}} 和 {{cssxref(":default")}}：分别针对被选中的复选框和单选按钮，处于不确定的状态（既不被选中也不被选中），以及页面加载时默认选择的选项（例如，一个设置了 `checked` 属性的 [`<input type="checkbox">`](/zh-CN/docs/Web/HTML/Element/input/checkbox)，或者一个设置了 `selected` 属性的 [`<option>`](/zh-CN/docs/Web/HTML/Element/option) 元素）。
```

File: files/zh-cn/learn/forms/ui_pseudo-classes/index.md [hunk 2]
```diff
@@ -0,0 +1,620 @@
...
+    </div>
+    <div>
+      <label for="pcode2" class="billing-label disabled-label">
+        邮政编码：
+      </label>
+      <input id="pcode2" name="pcode2" type="text" disabled required />
+    </div>
+  </fieldset>
+
+  <div><button>提交</button></div>
```
Comment (yin1999) [inline files/zh-cn/learn/forms/ui_pseudo-classes/index.md:368]: same, but we could localized the comment below :)

File: files/zh-cn/learn/forms/ui_pseudo-classes/index.md [hunk 3]
```diff
@@ -0,0 +1,620 @@
...
+
+使用 `readonly` 属性可将一个输入设置为只读。举个例子，设想一个确认页面，开发者将之前页面上填写的细节发送到这个页面，目的是让用户在一个地方检查所有细节，添加任何需要的最终数据，然后通过提交确认订单。在这一点上，所有最终的表单数据都可以一次性发送到服务器上。
+
+让我们来看看一个表单可能是什么样子的（见运行实例 [readonly-confirmation.html](https://mdn.github.io/learning-area/html/forms/pseudo-classes/readonly-confirmation.html)，其[源代码在这里](https://github.com/mdn/learning-area/blob/main/html/forms/pseudo-classes/readonly-confirmation.html)）。
+
+HTML 的一个片段如下，注意其中的 readonly 属性：
+
+```html
+<div>
+  <label for="name">姓名：</label>
```
Comment (yin1999) [inline files/zh-cn/learn/forms/ui_pseudo-classes/index.md:441]: +1

File: files/zh-cn/learn/forms/ui_pseudo-classes/index.md [hunk 4]
```diff
@@ -0,0 +1,620 @@
...
+  content: "Default";
+  color: white;
+  background-color: black;
+  padding: 5px 10px;
+  right: -65px;
+  top: -3px;
+}
+```
+
+这为页面加载时最初选择的项目提供了一个小小的“default”标签。注意这里我们使用的是一般的兄弟姐妹组合器（`~`），而不是相邻的兄弟姐妹组合器（`+`）——我们需要这样做，因为在源代码中，`<span>` 并不在 `<input>` 之后。
```
Comment (yin1999) [inline files/zh-cn/learn/forms/ui_pseudo-classes/index.md:554]:
```suggestion
这为页面加载时最初选择的项目提供了一个小小的“default”标签。注意这里我们使用的是通用兄弟组合器（`~`），而不是相邻兄弟组合器（`+`）——我们需要这样做，因为在源代码中，`<span>` 并不在 `<input>` 之后。
```

File: files/zh-cn/learn/forms/ui_pseudo-classes/index.md [hunk 5]
```diff
@@ -0,0 +1,620 @@
...
+
+这将在单选按钮上创建一个有趣的动画边框，希望它能表明你需要选择其中的一个！
+
+参见下方的实时结果：
+
+{{EmbedGHLiveSample("learning-area/html/forms/pseudo-classes/radios-checked-indeterminate.html", '100%', 200)}}
+
+> **备注：** 你也可以在 GitHub 的 [radios-checked-indeterminate.html](https://mdn.github.io/learning-area/html/forms/pseudo-classes/radios-checked-indeterminate.html) 上找到这个示例的运行实例，也可以参见[源代码](https://github.com/mdn/learning-area/blob/main/html/forms/pseudo-classes/radios-checked-indeterminate.html)。
+
+> **备注：** 你可以在 [`<input type="checkbox">`](/zh-CN/docs/Web/HTML/Element/input/checkboxes) 参考页上找到一个[涉及 `indeterminate` 状态的有趣示例](/zh-CN/docs/Web/HTML/Element/input/checkboxes)。
```
Comment (yin1999) [inline files/zh-cn/learn/forms/ui_pseudo-classes/index.md:589]:
```suggestion
> **备注：** 你可以在 [`<input type="checkbox">`](/zh-CN/docs/Web/HTML/Element/input/checkbox) 参考页上找到一个[涉及 `indeterminate` 状态的有趣示例](/zh-CN/docs/Web/HTML/Element/input/checkbox#中间状态复选框)。
```

**Review by yin1999 — COMMENTED** [review]
Could you also apply the suggestions list [here](https://pr13765.content.dev.mdn.mozit.cloud/zh-CN/docs/Learn/Forms/UI_pseudo-classes#_flaws) :)

File: files/zh-cn/learn/forms/ui_pseudo-classes/index.md [hunk 6]
```diff
@@ -0,0 +1,620 @@
...
+
+在下一节，我们将看一个更好的例子，使用 `:required` 表示必填字段，这也是对使用生成内容的挖掘。
+
+> **备注：** 你可能不会发现自己经常使用 `:optional` 这个伪类。表单控件默认是可选的，所以你可以将可选状态的样式当作默认样式，然后在必需的组件上添加样式。
+
+> **备注：** 如果一组同名的单选按钮中的一个单选按钮具有 `required` 属性，那么所有的单选按钮都将无效，直到有一个被选中，但只有分配了该属性的单选按钮才会真正匹配 {{cssxref(':required')}}。
+
+## 使用伪类生成内容
+
+在之前的文章中，我们已经看到了[生成内容](/zh-CN/docs/Web/CSS/CSS_Generated_Content)的用法，现在是一个更详细地谈论它的好时机。
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/ui_pseudo-classes/index.md:108]:
```suggestion
在之前的文章中，我们已经看到了[生成内容](/zh-CN/docs/Web/CSS/CSS_generated_content)的用法，现在是一个更详细地谈论它的好时机。
```

File: files/zh-cn/learn/forms/ui_pseudo-classes/index.md [hunk 7]
```diff
@@ -0,0 +1,620 @@
...
+正如我们在本模块前面的文章中所看到的，{{HTMLElement("input/radio", "单选按钮")}}和{{HTMLElement("input/checkbox", "复选框")}}可以被选中或不被选中。但也有一些其他的状态需要考虑：
+
+- {{cssxref(":default")}}：匹配在页面加载时默认选中的单选钮/复选框（即通过设置 `checked` 属性），这些匹配 {{cssxref(":default")}} 伪类，即使用户取消选中。
+- {{cssxref(":indeterminate")}}：当单选钮/复选框既没有被选中也没有被取消时，它们是*中间状态*，并将与 {{cssxref(":indeterminate")}} 伪类匹配。下文将阐述其细节。
+
+### :checked
+
+当单选钮或复选框被选中时，它们将被 {{cssxref(":checked")}} 伪类所匹配。
+
+最常见的用途是在复选框或单选按钮被选中时添加不同的样式，在这种情况下，已经使用 `appearance: none;` 删除了系统默认的样式，想自己重新建立样式。我们在上一篇文章中看到了这样的示例，当时我们谈到了[在单选钮/复选框上使用 `appearance: none`](/zh-CN/docs/Learn/forms/Advanced_form_styling#using_appearance_none_on_radioscheckboxes)。
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/ui_pseudo-classes/index.md:483]:
```suggestion
最常见的用途是在复选框或单选按钮被选中时添加不同的样式，在这种情况下，已经使用 `appearance: none;` 删除了系统默认的样式，想自己重新建立样式。我们在上一篇文章中看到了这样的示例，当时我们谈到了[在单选钮/复选框上使用 `appearance: none`](/zh-CN/docs/Learn/Forms/Advanced_form_styling#using_appearance_none_on_radioscheckboxes)。
```

**Review by yin1999 — APPROVED** [review]
Great. Let's landing this :+1:

**Pull request merged**